tls/s2n_config.c

```diff
@@ -214,27 +221,31 @@ static int s2n_config_build_domain_name_to_cert_map(struct s2n_config *config, s
 
 struct s2n_config *s2n_fetch_default_config(void)
 {
+    if (s2n_is_tls13_enabled()) {
```

Comment:
If the highest protocol version is TLS13, but the customer only wants to use TLS12, will this return an unexpected configuration?

Reply:
If we set cipher preferences and signature algorithms that don't work in tls1.3 when s2n_is_tls13_enabled() == true, things break. We will try to use tls1.3 (since connections will be initialized with *_protocol_version = S2N_TLS13), but won't have the right ciphers or signature schemes to do it. Also, the tls1.3 defaults still work for tls1.2.

Reply:
it feels like we should check for fips + tls13 combinations, but if this is fine if we assume that tls13 is not turned on fips